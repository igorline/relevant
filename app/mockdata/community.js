export const relevant = {
  _id: 'c00000000000000000000001',
  name: 'Relevant',
  slug: 'relevant',
  image: 'https://relevant-image.s3.amazonaws.com/9fgwyrd05_rL.jpg',
  description:
    'A community for curating and discussing links related to critical analysis of #technology, #society and #culture.',
  maxPostRank: 0.07075576196523553,
  maxUserRank: 0.16740002478323354,
  numberOfElements: 208,
  memberCount: 60,
  lastRewardFundUpdate: new Date(),
  postCount: 6,
  topPostShares: 1000,
  currentShares: 200,
  rewardFund: 0,
  currentPosts: 0,
  topics: ['technology', 'society', 'culture', 'politics', 'altwoke'],
  channels: [],
  betEnabled: true
};

export const crypto = {
  _id: 'c00000000000000000000002',
  name: 'Crypto',
  slug: 'crypto',
  image: '',
  description: 'Discussion of blockchain tech and ecosystem',
  rewardFund: 100 * 1e18,
  topics: [
    'governance',
    'layer1',
    'layer2',
    'whiter papers',
    'scaling',
    'projects',
    'reputation',
    'curation markets'
  ],
  channels: [],
  betEnabled: true
};

export const brandNew = {
  _id: 'c00000000000000000000003',
  name: 'New Community',
  slug: 'imnew',
  image: '',
  description: 'Discussion of blockchain tech and ecosystem',
  lastRewardFundUpdate: new Date(),
  topics: ['co community topic'],
  channels: [],
  superAdmins: ['alice'],
  admins: ['alice']
};

export const allCommunities = [relevant, crypto];

export const community = {
  communities: {
    relevant,
    crypto
  },
  userMemberships: [],
  userCommunities: [],
  list: ['relevant', 'crypto'],
  active: 'relevant'
};
