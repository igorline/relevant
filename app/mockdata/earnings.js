export const earningPaidout = {
  community: 'relevant',
  communityId: null,
  createdAt: new Date('January 31, 2019 04:01:01 EST').toString(),
  earned: 0.0918203703714779,
  estimatedPostPayout: 0,
  post: '5c45483709a5b05cd0646d7c',
  shares: 0,
  source: 'post',
  stakedTokens: 0,
  status: 'paidout',
  totalPostShares: 0,
  type: 'coins',
  updatedAt: '2019-02-05T05:42:29.684Z',
  user: '5c4267177f81360b10b4b196',
  __v: 0,
  _id: '5c5922452de74dc9e56eeebb'
};

export const earningPending1 = {
  _id: '5c5086662de74dc9e5f61b66',
  post: '5c3d07d0348c61cc78023449',
  user: '5c4267177f81360b10b4b196',
  __v: 0,
  community: 'relevant',
  communityId: '5ba92052a7c7a539ed46d76f',
  status: 'pending',
  type: 'coins',
  earned: 0,
  shares: 10,
  estimatedPostPayout: 600,
  totalPostShares: 100,
  stakedTokens: 60,
  source: 'post'
};

export const earningPending2 = {
  _id: '5c50866 42de74dc9e5f61b4d',
  post: '5c2ea2b7cdf48233c35f10cd',
  user: '5c4267177f81360b10b4b196',
  __v: 0,
  community: 'relevant',
  communityId: '5ba92052a7c7a539ed46d76f',
  status: 'pending',
  type: 'coins',
  earned: 0,
  shares: 10,
  estimatedPostPayout: 600,
  totalPostShares: 100,
  stakedTokens: 40,
  source: 'post'
};

export const earnings = {
  list: ['5c5086662de74dc9e5f61b66', '5c5086642de74dc9e5f61b4d'],
  entities: {
    '5c5086642de74dc9e5f61b4d': earningPending2,
    '5c5086662de74dc9e5f61b66': earningPending1
  },
  pending: ['5c5086662de74dc9e5f61b66', '5c5086642de74dc9e5f61b4d']
};
