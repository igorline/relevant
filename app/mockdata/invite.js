export const referral = {
  name: 'bob',
  type: 'referral'
};

export const referralWithEmail = {
  name: 'carol',
  type: 'referral',
  email: 'carol@carol.com'
};

export const adminReferral = {
  name: 'carol',
  type: 'admin',
  email: 'carol@carol.com'
};
