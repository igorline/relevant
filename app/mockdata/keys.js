export const keys = [
  {
    address: '0x627306090abaB3A6e1400e9345bC60c78a8BEf57',
    key: 'c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3'
  },
  {
    address: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
    key: 'ae6ae8e5ccbfb04590405997ee2d52d2b330726137b875053c36d94e974d162f'
  },
  {
    address: '0xC5fdf4076b8F3A5357c5E395ab970B5B54098Fef',
    key: '0dbbe8e4ae425a6d2687f1a7e3ba17bc98c673636790f1b8ad91193c05875ef1'
  }
];
