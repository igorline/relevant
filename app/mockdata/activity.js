import { user2 } from 'app/mockdata/user';
import { post2 } from 'app/mockdata/post';

export const activity = {
  totlaUsers: 3,
  amount: 10,
  byUser: user2, // TODO create user prop type validation
  post: post2, // TODO create post prop type validation
  type: 'upvote',
  createdAt: new Date()
};
