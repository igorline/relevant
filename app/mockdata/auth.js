import { user1 } from 'app/mockdata/user';

export const auth = {
  community: 'testCommunity',
  user: user1,
  isAuthenticated: true
};
