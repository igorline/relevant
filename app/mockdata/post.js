import { user2, user3 } from 'app/mockdata/user';

const IMAGES = [
  'https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/cute-cat-captions-1563551865.jpg',
  'https://i.pinimg.com/originals/5e/b4/49/5eb449b181999dabed59a857503031ad.jpg',
  'https://images.unsplash.com/photo-1497752531616-c3afd9760a11?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
  'https://i.pinimg.com/564x/e6/44/ea/e644eacfa75ecb5cb73e220bb0c7ef01.jpg'
];

export const post1 = {
  _id: '000000000000000000000001',
  user: 'b00000000000000000000000',
  title: 'postTitle',
  embeddedUser: user2,
  tags: ['tag11', 'tag12'],
  body: 'awesome post #1!',
  postDate: new Date('January 31, 2019 04:01:01 EST'),
  payoutTime: new Date('February 3, 2019 04:01:01 EST'),
  data: { pagerank: 45, relevance: 45, payout: 18 * 10 ** 18 },
  url: 'https://example.com/testPost1',
  image: IMAGES[0],
  link: {
    image: IMAGES[0],
    url: 'https://example.com/testPost1',
    domain: 'link.domain1',
    title: 'postTitle1'
  },
  parentPost: '000000000000000000000003',
  type: 'post',
  community: 'relevant',
  communityId: 'c00000000000000000000001',
  domain: 'link.domain1'
};

export const post2 = {
  _id: '000000000000000000000002',
  user: 'c00000000000000000000000',
  title: 'postTitle2',
  embeddedUser: user3,
  tags: ['tag21', 'tag22'],
  body: 'awesome post #2!',
  postDate: new Date('January 31, 2019 04:02:02 EST'),
  payoutTime: new Date('February 3, 2019 04:02:02 EST'),
  data: { pagerank: 45, payout: 18 * 10 ** 18 },
  url: 'https://example.com/testPost2',
  image: IMAGES[1],
  link: {
    image: IMAGES[1],
    url: 'https://example.com/testPost2',
    domain: 'link.domain2',
    title: 'postTitle2'
  },
  parentPost: '000000000000000000000003',
  parentComment: '000000000000000000000001',
  type: 'comment',
  community: 'relevant',
  communityId: 'c00000000000000000000001',
  domain: 'link.domain2'
};

export const linkPost1 = {
  _id: '000000000000000000000003',
  title: 'An Awesome Link #1 posted in both relevant and crypto',
  tags: ['tag31', 'tag32'],
  postDate: new Date(),
  payoutTime: new Date(),
  url: 'https://example.com/linkPost1',
  image: IMAGES[0],
  type: 'link',
  community: 'relevant',
  communityId: 'c00000000000000000000001',
  altCommunity: {
    community: 'crypto',
    communityId: 'c00000000000000000000002'
  },
  hidden: false,
  domain: 'link.domain3'
};

export const linkPost2 = {
  _id: '000000000000000000000004',
  title: 'An Awesome Link #2',
  tags: ['tag31', 'tag32'],
  postDate: new Date(),
  payoutTime: new Date(),
  url: 'https://example.com/linkPost2',
  image: IMAGES[1],
  type: 'link',
  community: 'relevant',
  communityId: 'c00000000000000000000001',

  hidden: false,
  domain: 'link.domain4'
};

export const linkPost3 = {
  _id: '000000000000000000000005',
  title: 'An Awesome Link #3 posted in crypto',
  tags: ['tag51', 'tag52'],
  postDate: new Date(),
  payoutTime: new Date(),
  url: 'https://example.com/linkPost3',
  image: IMAGES[2],
  type: 'link',
  community: 'crypto',
  communityId: 'c00000000000000000000002',
  hidden: false,
  domain: 'link.domain5'
};

const tomorrow = new Date();
tomorrow.setDate(tomorrow.getDate() + 1);

export const linkPost4 = {
  _id: '000000000000000000000006',
  title: 'An Awesome Link #4 - Pending payout',
  tags: ['tag51', 'tag52'],
  postDate: new Date(),
  payoutTime: tomorrow,
  url: 'https://example.com/linkPost4',
  image: IMAGES[3],
  type: 'link',
  community: 'crypto',
  communityId: 'c00000000000000000000002',
  hidden: false,
  domain: 'link.domain5'
};

export const linkPost5 = {
  _id: '000000000000000000000007',
  title: 'An Awesome Link #6',
  tags: ['tag51', 'tag52'],
  postDate: new Date(),
  payoutTime: new Date(),
  url: 'https://example.com/linkPost5',
  image: IMAGES[0],
  type: 'link',
  community: 'crypto',
  communityId: 'c00000000000000000000002',
  hidden: false,
  domain: 'link.domain6'
};

export const createPost = {
  url: 'https://example.com/testPost1',
  image: IMAGES[0],
  tags: ['tag11', 'tag12'],
  body: 'awesome post #1!',
  title: 'postTitle',
  description: 'my amazing post',
  mentions: ['test', 'user1'],
  domain: 'example.com'
};

export const postsState = {
  posts: {
    [post1._id]: post1,
    [post2._id]: post2,
    [linkPost1._id]: linkPost1,
    [linkPost2._id]: linkPost2
  }
};
