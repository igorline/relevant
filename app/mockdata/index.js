import * as activity from 'app/mockdata/activity';
import * as auth from 'app/mockdata/auth';
import * as community from 'app/mockdata/community';
import * as earnings from 'app/mockdata/earnings';
import * as post from 'app/mockdata/post';
import * as user from 'app/mockdata/user';
import * as twitter from 'app/mockdata/twitter';

export { activity, auth, community, earnings, post, user, twitter };
