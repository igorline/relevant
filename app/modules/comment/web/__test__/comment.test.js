import { act } from 'react-test-renderer';
import { CommentFormComponent } from 'modules/comment/web/commentForm.component';
import 'jest-styled-components';
import { post, user } from 'mockdata';
import { withRenderer } from 'jest.setup';

const state = {
  auth: { user: user.user1, isAuthenticated: true },
  navigation: { screenSize: 0 }
};

const props = {
  nestingLevel: 0,
  buttonText: 'Comment',
  parentPost: post.linkPost1,
  comment: post.post1,
  additionalNesting: 0,
  history: {}
};

test('Snapshot Comment Form Web', async () => {
  const tree = withRenderer({
    state,
    props,
    Component: CommentFormComponent
  });

  const textArea = tree.root.find(el => el.type === 'textarea');

  act(() => {
    textArea.props.onFocus();
    textArea.props.onChange({ target: { value: 'test input' } });
  });

  expect(textArea.props.value).toBe('test input');
  expect(tree.toJSON()).toMatchSnapshot();

  const buttons = tree.root.findAll(el => el.type === 'button');
  const cancel = buttons.find(b => b.children[0] === 'Cancel');

  act(() => {
    textArea.props.onChange({ target: { value: '' } });
    cancel.props.onMouseDown({});
  });
  expect(textArea.props.value).toBe('');
});
