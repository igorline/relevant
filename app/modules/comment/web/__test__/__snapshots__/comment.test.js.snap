// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Comment Form Web 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  margin: 0rem 0rem 1.25rem 0rem;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  background: #0000ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3rem;
  min-width: 9rem;
  padding: 0 1rem;
  margin: 0;
  font-size: 0.75rem;
  color: hsl(0,0%,100%);
  font-family: HelveticaNeue-Medium;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: transparent;
  padding: null;
  min-width: 0.5rem;
  margin-right: 0.5rem;
  color: hsl(0,0%,50%);
  -webkit-text-decoration-color: hsl(0,0%,50%);
  text-decoration-color: hsl(0,0%,50%);
  min-width: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c5:hover {
  box-shadow: 0 2px 6px hsl(240,0%,70%);
}

.c6 {
  background: #0000ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3rem;
  min-width: 9rem;
  padding: 0 1rem;
  margin: 0;
  font-size: 0.75rem;
  color: hsl(0,0%,100%);
  font-family: HelveticaNeue-Medium;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: null;
  min-width: 0.5rem;
  min-width: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c6:hover {
  box-shadow: 0 2px 6px hsl(240,0%,70%);
}

.c3 {
  font-size: 0.875rem;
  line-height: 1.2249999999999999rem;
  font-family: HelveticaNeue;
  color: hsl(0,0%,0%);
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-color: hsl(0,0%,87%);
  border-width: 1px;
  border-style: solid;
  padding: 1rem 1rem;
  min-width: 0;
  padding-left: 1rem;
  min-height: 4rem;
}

.c3:focus {
  outline: 1px solid #0000ff;
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "horizontal",
      "WebkitFlexDirection": "row",
      "WebkitFlexGrow": 1,
      "backgroundColor": "rgba(250,250,250,1.00)",
      "display": "flex",
      "flexDirection": "row",
      "flexGrow": 1,
      "msFlexDirection": "row",
      "msFlexPositive": 1,
      "paddingTop": "2rem",
      "position": "relative",
    }
  }
>
  <div
    className="c0"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="rta  "
          style={
            Object {
              "display": "flex",
              "flex": "auto",
              "flexDirection": "column",
            }
          }
        >
          <textarea
            autoFocus={false}
            className="rta__textarea c3"
            flex={1}
            m={0}
            minheight="4rem"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onScroll={[Function]}
            onSelect={[Function]}
            pl={2}
            placeholder="Enter comment..."
            rows={2}
            style={
              Object {
                "flex": "auto",
                "height": 0,
              }
            }
            value="test input"
          />
        </div>
      </div>
    </div>
    <div
      className="c4"
    >
      <button
        className="c5"
        disabled={false}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        Cancel
      </button>
      <button
        className="c6"
        disabled={false}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        Comment
      </button>
    </div>
  </div>
</div>
`;
